cmd/template/nodepool/flag.go

```diff
@@ -82,6 +86,8 @@ func (f *flag) Init(cmd *cobra.Command) {
 
 	// Azure only.
 	cmd.Flags().StringVar(&f.AzureVMSize, flagAzureVMSize, "Standard_D4s_v3", "Azure VM size to use for workers, e.g. 'Standard_D4s_v3'.")
+	cmd.Flags().BoolVar(&f.AzureUseSpotVms, flagAzureUseSpotVMs, false, "Whether to use Spot VMs for this Node Pool.")
+	cmd.Flags().Float32Var(&f.AzureSpotMaxPrice, flagAzureSpotMaxPrice, 0, "Max hourly price in USD to pay for one spot VM.")
```

Comment:
Should we default this to `-1`, and specify that by default it uses on-demand pricing as max?

Reply:
The default is really not important as it gets set to -1 if it's not > 0 in modepool/provider/azure.go
Adjusted description